src/service_info.rs

```diff
@@ -366,6 +366,18 @@ impl ServiceInfo {
             .cloned()
             .unwrap_or(ServiceStatus::Unknown)
     }
+
+    /// Consumes self and returns a resolved service, i.e. a lite version of `ServiceInfo`.
+    pub fn as_resolved_service(self) -> ResolvedService {
+        ResolvedService {
+            ty_domain: self.ty_domain,
```

Comment:
Another question:  what if this service info has some `sub_domain` ?  Currently, `ServiceInfo` stores only the regular service type & domain in `ty_domain`, and the full subtype & domain in `sub_domain`.   Do we want the same thing in `ResolvedService`?

Reply:
I think we should make `sub_domain` public too. If we are consuming `ServingInfo`, we can move that information into `ResolvedService`. We do not know the kind of use cases out there, perhaps someone needs that information

Reply:
`ty_domain` and `sub_domain` was not public, but they are indeed accessible via a public API (getter). I agree with you, I think it's better to include `sub_domain` in `ResolvedService` as well.

Reply:
With public, I meant accessible via public interface :P

Reply:
I've updated the diff and added `sub_ty_domain`. No longer a draft ;-).  Let me know if this patch works for you. Thanks!